### ISSUE TITLE
Calendar "blinks" on day click, if CalendarView has margins set

### ISSUE THREAD

**Issue by gajicm93** [kizitonwose/CalendarView#6]
This is not critical and not an issue for me currently, but if I set margins on a Calendar, start and end in my case, the whole CalendarView blinks every time I click on a day.

**Comment by gajicm93** [comment]
Though since CalendarView is actually a RecyclerView, it's much better to use padding with clipToPadding set to false.. that way padding won't cut of the view while scrolling, while margins would.

**Comment by kizitonwose** [comment]
Does using paddings instead of margins fix the blinking issue?

**Comment by kizitonwose** [comment]
Also, note that calling `notifyCalendarChanged()` will cause the calendar to blink. You're better off with `notifyDateChanged()`

**Comment by gajicm93** [comment]
It doesn't blink with padding, but padding doesn't also doesn't do anything with the view, it's still visible all across the screen. At least with start/end padding on a paged calendar.

Anyways, this is not really an issue for me, so you can just look into margin blinking if you have time, no pressure.

**Comment by kizitonwose** [comment]
To try to replicate this issue, I added 40dp start and end margins to the CalendarView in example 3. However, it does not blink when I click on a date.
You can see the CalendarView with margins below:

<img width="280" alt="Screenshot 2019-06-06 at 2 36 30 PM" src="https://user-images.githubusercontent.com/15170090/59037288-e4e99380-8868-11e9-8be5-6072bf6f4048.png">

**Comment by gajicm93** [comment]
Ok just ignore it then, thank you.

**Comment by Widuka** [comment]
@gajicm93 I have a similar problem. It appear when the main layout was ConstraintLayout when I change the main layout to for example FrameLayout blink bug didn't appear. I try also wrapping -> main ConstraintLayout -> FrameLayout -> CalenadarView. Using GuideLines. However, blink bug still appears. @kizitonwose try to set margin in example 1 where CalendarView is in ConstraintLayout. It will start blinking on day selection.

**Comment by kizitonwose** [comment]
I can confirm that this truly happens in certain cases. What I have observed is that for some reason, when the `CalendarView` is in a ConstraintLayout with some margins and you call `notifyDateChanged` or `notifyDayChanged`, the payload which contains the specific day to be reloaded is no longer passed to the `onBindViewHolder` method in the adapter hence the entire month is reload which gives the blink effect.

**I have also come up with a workaround for this:** 
I noticed that setting the `hasFixedSize` property in the `RecyclerView` to `true` fixes this problem. To do this, simply call: `calendarView.setHasFixedSize(true)`

This is a problem with `RecyclerView` itself which is CalendarView's base class.
